### PR TITLE
Convert some types from class to struct.

### LINKED ISSUE #79: Fix struct boxing and unboxing from unions
There seem to be some issues with the new struct feature not boxing and unboxing properly from union types. I've seen some runtime crashes related to this that I will fix so we can get the feature working properly.

### PR DESCRIPTION
**Pull request by jemc**
This PR converts a few types from `:class` to `:struct` and clears up some issues that were encountered while doing so. See individual commit messages for details.

Resolves #79.

### CODE REVIEW

File: src/prelude/string.savi [hunk 1]
```diff
@@ -442,8 +442,7 @@
     )
     --res
 
-// TODO: Convert into `:struct`
-:class val StringPair
+:struct val StringPair
```
Comment (mneumann) [inline src/prelude/string.savi:445]: I am looking forward to see a generic `Pair(T)` type.

**Review by mneumann — APPROVED** [review]
looks great

File: packages/collections/map.savi [hunk 2]
```diff
@@ -73,18 +73,15 @@
     value_alias V'aliased = value // TODO: remove the V'aliased explicit type?
     try (
       index = @_search(key)
-      entry = @_array[index]!
+      entry = @_array[index]! <<= _KV(K, V).new(--key, --value)
```
[the remark this answers is not in the document]
Comment (jemc) [reply packages/collections/map.savi]: Done in 3914f1f4cad84c521a831ee894b69e2a106bc866.

File: src/prelude/string.savi (hunk 1 above)
Comment (jemc) [reply src/prelude/string.savi:445]: Added in https://github.com/savi-lang/savi/pull/122

**Pull request merged**